src/modules/cpu/kernel/down_mixing.hpp

```diff
@@ -0,0 +1,148 @@
+                                  RpptDescPtr srcDescPtr,
+                                  Rpp32f *dstPtr,
+                                  RpptDescPtr dstDescPtr,
+                                  Rpp32s *srcLengthTensor,
+                                  Rpp32s *channelsTensor,
+                                  bool normalizeWeights,
+                                  rpp::Handle& handle)
+{
+    Rpp32u numThreads = handle.GetNumThreads();
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < srcDescPtr->n; batchCount++)
+    {
+        Rpp32f *srcPtrTemp = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        Rpp32f *dstPtrTemp = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32s channels = channelsTensor[batchCount];
+        Rpp32s samples = srcLengthTensor[batchCount];
+        bool flag_avx = 0;
+
+        if(channels == 1)
+        {
+            // No need of downmixing, do a direct memcpy
+            memcpy(dstPtrTemp, srcPtrTemp, (size_t)(samples * sizeof(Rpp32f)));
+        }
+        else
+        {
+            Rpp32f *weights = static_cast<Rpp32f *>(malloc(channels * sizeof(Rpp32f)));
+            std::fill(weights, weights + channels, 1.f / channels);
+
+            if(normalizeWeights)
+            {
+                // Compute sum of the weights
+                Rpp32f sum = 0.0;
+                for(int i = 0; i < channels; i++)
+                    sum += weights[i];
+
+                // Normalize the weights
+                Rpp32f invSum = 1.0 / sum;
+                for(int i = 0; i < channels; i++)
+                    weights[i] *= invSum;
+            }
+
+            Rpp32s channelIncrement = 4;
+            Rpp32s alignedChannels = (channels / 4) * 4;
+
+            if(channels > 6)
+            {
+                flag_avx = 1;
+                channelIncrement = 8;
+                alignedChannels = (channels / 8) * 8;
+            }
+
+            // use weights to downmix to mono
+            for(int64_t dstIdx = 0; dstIdx < samples; dstIdx++)
+            {
+                Rpp32s channelLoopCount = 0;
+                if(flag_avx)
+                {
+                    __m256 pDst = _mm256_setzero_ps();
+                    for(; channelLoopCount < alignedChannels; channelLoopCount += channelIncrement)
+                    {
+                        __m256 pSrc, pWeights;
+                        pWeights = _mm256_setr_ps(weights[channelLoopCount], weights[channelLoopCount + 1], weights[channelLoopCount + 2], weights[channelLoopCount + 3],
+                                weights[channelLoopCount + 4], weights[channelLoopCount + 5], weights[channelLoopCount + 6], weights[channelLoopCount + 7]);
+                        pSrc = _mm256_loadu_ps(srcPtrTemp);
+                        pSrc = _mm256_mul_ps(pSrc, pWeights);
+                        pDst = _mm256_add_ps(pDst, pSrc);
```

Comment:
Not using fmadd since rppt_audio cpp file is alone built without the fma flag?

Reply:
Yes that is the reason